.github/workflows/metadata-check.sh

```diff
@@ -0,0 +1,73 @@
+#! /bin/bash
+
+# Check the 'cargo metadata' for various requirements
+
+set -e
+
+META=$(mktemp)
+trap 'rm -rf -- "${META}"' EXIT
+
+cargo metadata --locked --format-version 1 > "${META}"
+
+get_version() {
+  local package="${1}"
+  jq -r '.packages[] | select(.name == "'"${package}"'") | .version' "${META}"
+}
+
+get_msrv() {
+  local package="${1}"
+  jq -r '.packages[] | select(.name == "'"${package}"'") | .rust_version' "${META}"
+}
+
+# check that the cxx packages all have the same version
+check_cxx_versions() {
+  local cxx_version=$(get_version "cxx")
+  local cxx_build_version=$(get_version "cxx-build")
+  local cxxbridge_cmd_version=$(get_version "cxx-build")
+  local cxxbridge_flags_version=$(get_version "cxxbridge-flags")
+  local cxxbridge_macro_version=$(get_version "cxxbridge-macro")
+
+  ok=true
+  echo "Found cxx version ${cxx_version}"
+  if [ "${cxx_version}" != "${cxx_build_version}" ]; then
+    echo "Found differing cxx-build version ${cxx_build_version}"
+    ok = false
+  fi
+  if [ "${cxx_version}" != "${cxxbridge_cmd_version}" ]; then
+    echo "Found differing cxxbridge-cmd version ${cxxbridge_cmd_version}"
+    ok = false
+  fi
+  if [ "${cxx_version}" != "${cxxbridge_flags_version}" ]; then
```

Comment:
This is already enforced by https://github.com/dtolnay/cxx/blob/1.0.124/Cargo.toml#L31.